BeautifulSoupScript_edits.py

```diff
@@ -0,0 +1,213 @@
+    'https://oi.uchicago.edu/research/oimp',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-nubian-expedition-oine',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-publications-oip',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-seminars-ois',
+    'https://oi.uchicago.edu/research/publications/studies-ancient-oriental-civilization-saoc',
+    'https://oi.uchicago.edu/research/publications/lost-egypt',
+    'https://oi.uchicago.edu/research/publications/miscellaneous-publications'
+]
+
+
+OIC_ROOT = 'https://oi.uchicago.edu'
+
+def main(journal_key=0):
+    #Looping through each catalog to add the download links and metadata to a dictionary
+    for i, url in enumerate(request_urls):
+        try:
+            if i == 2:
+                continue
+            else:
+                elem = fetchPageHTML(url)
+        except Exception as e:
+            print(e)
+            continue
+
+        soup = BeautifulSoup(elem.text, 'lxml')   #Parsing the catalog's webpage using BeautifulSoup
+
+        #Parsing for dictionary catalogs
+        if i == 1 or i == 3 or i == 15:
+
+            catContainer = soup.find(class_='catalog')
+
+            pubList = catContainer.find('tbody')
+
+            pubEntry = pubList.find_all('tr')
+
+            for pub in pubEntry:
+                print(url)
+
+                print(parse_Pub_Dict(pub))
+
+        #Parsing for journal catalogs
+        else:
+            journal_key += 1
+
+            catContainer = soup.find(class_='content-inner-left')
+
+            pubList = catContainer.find('ul')
+
+            #Looping through unordered list of publications
+            for pub in pubList.children:
+                #Testing to make sure pub is a <li> Tag and not an empty string
+                if not isinstance(pub, Tag):
+                    continue
+                metadata = parse_Pub_Jour(pub, journal_key)
+
+                if metadata is None:
+                    continue
+
+                downloadLink = pub.find('a', class_= "publication ss-standard ss-download btn")
+
+                if downloadLink is None:
+                    continue
+
+                metadata['url'] = downloadLink.get('href', None)
+                print(metadata)
+        '''
+        #Serializing download_dict to create a JSON file
+        with open("data_file.json", "w") as write_file:
+            json.dump(download_dict, write_file)
+        '''
+#Parsing the journal and miscelleanous publications for metadata at the bottom of webpage
+def parse_Pub_Jour(pub, journal_key):
+    pubLinks = pub.find_all('a')
+    #Link should only be the link to the publication webpage
+    detailLink = pubLinks[0].get('href', None)
+
+    detailURL = '{}{}'.format(OIC_ROOT, detailLink)
+
+    print(detailURL)
+
+    try:
+        pubElem = fetchPageHTML(detailURL)
+    except ConnectionError or HTTPError:
```

Comment:
👍 I think it's really important to only explicitly handle exceptions we expect to get.